openage/convert/gamedata/empiresdat.py

```diff
@@ -70,27 +71,76 @@ class for fighting and beating the compressed empires2*.dat
 
         # graphic data
         (READ, "graphic_count", "uint16_t"),
-        (READ, "graphic_offsets", "int32_t[graphic_count]"),
+        (READ, "graphic_offsets", "uint32_t[graphic_count]"),
         (READ_EXPORT, "graphics", SubdataMember(
             ref_type  = graphic.Graphic,
             length    = "graphic_count",
             offset_to = ("graphic_offsets", lambda o: o > 0),
         )),
-        (READ_UNKNOWN, "rendering_blob", "uint8_t[138]"),
 
         # terrain data
+        (READ, "map_pointer", "int32_t"),
+        (READ_UNKNOWN, None, "int32_t"),
+        (READ, "map_width", "int32_t"),
+        (READ, "map_height", "int32_t"),
+        (READ, "world_width", "int32_t"),
+        (READ, "world_height", "int32_t"),
+        (READ_EXPORT,  "tile_sizes", SubdataMember(
+            ref_type=terrain.TileSize,
+            length=19,      # number of tile types
+        )),
+        (READ_UNKNOWN, None, "int16_t"),
         (READ_EXPORT,  "terrains", SubdataMember(
             ref_type=terrain.Terrain,
-            length="terrain_count",
+            length=42,      # 42 terrains are stored, but less are used.
```

Comment:
because terrain_count no longer equals the number of read terrains. `terrain_count` is the number of used terrains, but i think 2 more are following with zeroed out values, which are now also read.